src/Microsoft.Extensions.Configuration.AzureAppConfiguration/Extensions/StringExtensions.cs

```diff
@@ -1,10 +1,67 @@
 ﻿// Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 //
+using Microsoft.Extensions.Configuration.AzureAppConfiguration.AzureKeyVault;
+using Microsoft.Extensions.Configuration.AzureAppConfiguration.FeatureManagement;
+using System.Collections.Generic;
+using System.Linq;
+using System;
+using System.Net.Mime;
+
 namespace Microsoft.Extensions.Configuration.AzureAppConfiguration.Extensions
 {
     internal static class StringExtensions
     {
+        private static readonly IEnumerable<string> ExcludedJsonContentTypes = new[]
+        {
+            FeatureManagementConstants.ContentType,
+            KeyVaultConstants.ContentType
+        };
+
+        public static bool IsJsonContentType(this string contentType)
+        {
+            if (string.IsNullOrWhiteSpace(contentType))
+            {
+                return false;
+            }
+
+            string acceptedMainType = "application";
+            string acceptedSubType = "json";
+            string mediaType;
+
+            try
+            {
+                mediaType = new ContentType(contentType.Trim()).MediaType;
+            }
+            catch (FormatException)
+            {
+                return false;
+            }
+            catch (IndexOutOfRangeException)
+            {
+                // Bug in System.Net.Mime.ContentType throws this if contentType is "xyz/"
+                // https://github.com/dotnet/runtime/issues/39337
+                return false;
+            }
```

Comment:
I think this should be `TryParseContentType` Then whatever we do with a content type object can come after.